### PR TITLE
Apply k8s 1.18 clientgo changes in cluster-api

### PR DESCRIPTION
**Pull request by tomahawk28**
Apply k8s 1.18.5 into cluster-api after forks. (same as cluster-runtime to 0.6.1 to follow upstream changes)

### CODE REVIEW

**Review by rossf7 — APPROVED** [review]
LGTM but please also ask for reviews from the rest of the WG.

As they know more about CAPI.

**Comment by tomahawk28** [comment]
Do you mean wg from cluster-api? right?

**Comment by tomahawk28** [comment]
I'm assigned you guys by the manual way since our WG have no available seat, pretty packed :)

File: controllers/remote/cluster_cache.go [hunk 1]
```diff
@@ -474,7 +474,7 @@ func healthCheckPath(sourceCfg *rest.Config, requestTimeout time.Duration, path
 		return err
 	}
 
-	_, err = restClient.Get().AbsPath(path).Timeout(requestTimeout).Do().Get()
+	_, err = restClient.Get().AbsPath(path).Timeout(requestTimeout).Do(context.TODO()).Get()
```
Comment (fiunchinho) [inline controllers/remote/cluster_cache.go]: do you know why it adds `context.TODO()` instead of `context.Background()`?

Comment (tomahawk28) [reply controllers/remote/cluster_cache.go]: I think I forgot to change this part, going to change it now.

File: controllers/remote/cluster_cache.go [hunk 2]
```diff
@@ -396,15 +396,15 @@ func (h *healthCheckInput) validate() {
...
+func (m *ClusterCacheTracker) healthCheckCluster(ctx context.Context, in *healthCheckInput) {
 	// populate optional params for healthCheckInput
 	in.validate()
 
 	unhealthyCount := 0
 
 	runHealthCheckWithThreshold := func() (bool, error) {
 		cluster := &clusterv1.Cluster{}
-		if err := m.client.Get(context.TODO(), in.cluster, cluster); err != nil {
+		if err := m.client.Get(context.Background(), in.cluster, cluster); err != nil {
```
Comment (nprokopic) [inline controllers/remote/cluster_cache.go]: `ctx` is available in this function.

```suggestion
		if err := m.client.Get(ctx, in.cluster, cluster); err != nil {
```

**Review by nprokopic — APPROVED** [review]
LGTM. Small nit, not that it matters that much, since we will be using only CRDs from the fork, and for a limited time period.

**Pull request merged**